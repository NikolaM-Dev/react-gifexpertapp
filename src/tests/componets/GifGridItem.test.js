import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../componets/GifGridItem';

describe('Pruebas en GifGridItem', () => {
  const title = 'Un titulo';
  const url = 'https://localhost/algo.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('Debe de mostrar correctamente <GifGridItem />', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de tener un parrafo con el ttitle', () => {
    const p = wrapper.find('p').text().trim();

    expect(p).toBe(title);
  });

  test('Debe de tener la imagen al url y alt de los props', () => {
    const img = wrapper.find('img');

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('Debe tener la class animate__fadeIn', () => {
    const div = wrapper.find('div');
    const include = div.prop('className').includes('animate__fadeIn');

    expect(include).toBe(true);
  });
});
