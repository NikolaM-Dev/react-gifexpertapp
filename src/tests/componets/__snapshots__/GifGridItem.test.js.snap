// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe de mostrar correctamente <GifGridItem /> 1`] = `
<div
  className="card animate__animate danimate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
